core/src/main/java/jeeves/server/context/ServiceContext.java

```diff
@@ -52,21 +53,179 @@
 
 /**
  * Contains the context for a service execution.
+ *
+ * When creating a ServiceContext you are responsible for manging its use on the current thread and any cleanup:
+ * <pre><code>
+ * try {
+ *    context = serviceMan.createServiceContext("AppHandler", appContext);
+ *    context.setAsThreadLocal();
+ *    ...
+ * } finally {
+ *    context.clearAsThreadLocal();
+ *    context.clear();
+ * }</code></pre>
+ *
+ * @see ServiceManager
  */
-public class ServiceContext extends BasicContext {
+public class ServiceContext extends BasicContext implements AutoCloseable {
```

Comment:
This looks good and is the key change, the above javadoc code example could include a try-with-resources example.

```
 * ServiceContext is auto closable for use with try-with-resource syntaxt:
 * <pre></code>
 * try (ServiceContext context = serviceMan.createServiceContext("AppHandler", appContext)){
 *     ...
 * }
 * </code></pre>
```

There are other javadocs with this example, but it is nice to call it out in the class description.